pkg/client/cli/connect/connector.go

```diff
@@ -288,21 +288,18 @@ func newUserDaemon(conn *grpc.ClientConn, daemonID *daemon.Identifier) *daemon.U
 func EnsureUserDaemon(ctx context.Context, required bool) (context.Context, error) {
 	var err error
 	var ud *daemon.UserClient
-	defer func() {
-		if err == nil && required && !ud.Containerized() {
-			// The RootDaemon must be started if the UserDaemon was started
-			err = ensureRootDaemonRunning(ctx)
-		}
-	}()
-
 	if ud = daemon.GetUserClient(ctx); ud != nil {
 		return ctx, nil
 	}
 	if ctx, ud, err = launchConnectorDaemon(ctx, client.GetExe(ctx), required); err != nil {
 		return ctx, err
 	}
 	ctx = daemon.WithUserClient(ctx, ud)
-	return ctx, nil
+	if required && !ud.Containerized() {
+		// The RootDaemon must be started if the UserDaemon was started
+		err = ensureRootDaemonRunning(ctx)
+	}
+	return ctx, err
```

Comment:
I'm not in favor of this fix, because a return on (former) lines 299 or 302 will no longer ensure that the root daemon is running.